src/frontend/src/lib/utils/format.utils.ts

```diff
@@ -68,6 +68,30 @@ export const formatNanosecondsToDate = (nanoseconds: bigint): string => {
 	return date.toLocaleDateString('en', DATE_TIME_FORMAT_OPTIONS);
 };
 
+export const formatSecondsToNormalizedDate = ({
+	seconds,
+	currentDate
+}: {
+	seconds: number;
+	currentDate?: Date;
+}): string => {
+	const date = new Date(seconds * 1000);
+	const today = currentDate ?? new Date();
+	const daysDifference = Math.ceil((date.getTime() - today.getTime()) / (1000 * 60 * 60 * 24));
+
+	if (Math.abs(daysDifference) < 2) {
+		return new Intl.RelativeTimeFormat('en', { numeric: 'auto' }).format(daysDifference, 'day');
+	}
+
+	// Same year, return day and month name
+	if (date.getFullYear() === today.getFullYear()) {
+		return date.toLocaleDateString('en', { day: 'numeric', month: 'long' });
```

Comment:
I just read the following on [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toLocaleDateString)

> When the method is called many times with the same arguments, it is better to create a [Intl.DateTimeFormat](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DateTimeFormat) object and use its [format()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DateTimeFormat/format) method, because a DateTimeFormat object remembers the arguments passed to it and may decide to cache a slice of the database, so future format calls can search for localization strings within a more constrained context.

Maybe for this implementation and our existing implementation, we should avoid to recreate always `new Intl` or `date.toLocal...` to improve the performance?

**NOT** for this PR, just sharing and asking.

Reply:
yes, it makes sense! i add a TODO and put it in my backlog

Reply:
Done with :

- PR #3603